subsys/net/lib/fota_download/src/fota_download.c

```diff
@@ -300,6 +300,42 @@ static bool is_ip_address(const char *host)
 	return false;
 }
 
+int fota_download_s0_active_get(bool *const s0_active)
+{
+	if (!s0_active) {
+		return -EINVAL;
+	}
+
+#ifdef PM_S1_ADDRESS
+#ifdef CONFIG_TRUSTED_EXECUTION_NONSECURE
+	if (spm_s0_active(PM_S0_ADDRESS, PM_S1_ADDRESS, s0_active)) {
+		return -ENODEV;
+	}
+
+#else /* CONFIG_TRUSTED_EXECUTION_NONSECURE */
+	const struct fw_info *s0;
+	const struct fw_info *s1;
+
+	s0 = fw_info_find(PM_S0_ADDRESS);
+	if (s0 == NULL) {
+		return -EFAULT;
+	}
+
+	s1 = fw_info_find(PM_S1_ADDRESS);
+	if (s1 == NULL) {
+		/* No s1 found, s0 is active */
+		*s0_active = true;
+	} else {
+		/* Both s0 and s1 found, check who is active */
+		*s0_active = s0->version >= s1->version;
```

Comment:
I'm suspicious of this.  How do you know that s0 is active and not s1, if the version is equal?

Reply:
that is how the bootloader handles the slots/versions:
https://github.com/nrfconnect/sdk-nrf/blob/bc2e9b452cb49093eff50e25a8fc2a9d8ce63a9a/samples/bootloader/src/main.c#L95